core/deployment/src/main/java/io/quarkus/deployment/util/UriNormalizationUtil.java

```diff
@@ -91,14 +91,18 @@ public static URI toURI(String path, boolean trailingSlash) {
      * </ul>
      *
      * @param base URI to resolve relative paths. Use {@link #toURI(String, boolean)} to construct this parameter.
-     * 
+     *
      * @param segment Relative or absolute path
      * @param trailingSlash true if resulting URI must end with a '/'
      * @throws IllegalArgumentException if the path contains invalid characters or path segments.
      */
     public static URI normalizeWithBase(URI base, String segment, boolean trailingSlash) {
         if (segment == null || segment.trim().isEmpty()) {
-            return base;
```

Comment:
In this case, the method would have returned `/app/` for instance even if `trailingSlash` was false, which breaks the contract of a few of your methods that have a javadoc saying the path will never end with a `/`. I also had to fix a test that was incorrect.